src/main/java/org/mariadb/jdbc/internal/util/RedirectionInfo.java

```diff
@@ -0,0 +1,126 @@
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list
+ * of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice, this
+ * list of conditions and the following disclaimer in the documentation and/or
+ * other materials provided with the distribution.
+ *
+ * Neither the name of the driver nor the names of its contributors may not be
+ * used to endorse or promote products derived from this software without specific
+ * prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS  AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+ * OF SUCH DAMAGE.
+ *
+ */
+
+package org.mariadb.jdbc.internal.util;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.mariadb.jdbc.HostAddress;
+
+public class RedirectionInfo {
+	//Redirection Info format: Location: mysql://[%s]:%u/?user=%s&ttl=%u\n
+
+    private final HostAddress host;
+    private final String user;
+    private final int ttl;
+
+    public RedirectionInfo(HostAddress host, String user, int ttl) {
+        this.host 	= host;
+        this.user 	= user;
+        this.ttl 	= ttl;
+    }
+
+    public HostAddress getHost() {
+        return host;
+    }
+
+    public String getUser() {
+        return user;
+    }
+    
+    public int getTTL() {
+    	return ttl;
+    }
+
+    /**
+    * Parse redirection info from a message return by server.
+    *
+    * @param msg  the string which may contain redirection information.
+    * @return RedirectionInfo host and user for redirection.
+    */
+    public static RedirectionInfo parseRedirectionInfo(String msg) {
+        /**
+         * Get redirected server information contained in OK packet.
+         * Redirection string format:
+         * Location: mysql://[%s]:%u/?user=%s&ttl=%u\n
+         */
+    	String host = "";
+    	String user = "";
+    	int port = -1;
+    	int ttl = -1;
+        try {
+
+            Pattern INFO_PATTERN =
+            		Pattern.compile("^Location: mysql://\\[([^\\]:]+)\\]:([0-9]+)/\\?user=([^&]+)&ttl=([0-9]+)\\n");
```

Comment:
Does this need any flexibility for future server-side changes? E.g., allow `ttl` "parameter" to be omitted, or allow additional parameters to be added?

Although, arguably, if additional parameters were added by the server, it might be safest to fail to parse the string on the client, as we can't guarantee that it would safe to simply ignore any extra parameters.

Reply:
From server side's change, the field ttl is always, so I think the code is safe. About the flexibility,  I think maybe we we can handle it in future if new field and related logic is introduced,  right?